index.html

```diff
@@ -4292,6 +4306,454 @@ <h3>Proof Formats</h3>
+that is associated with a sequence of bytes is valid. This algorithm takes a
+byte sequence (bytes <var>inputBytes</var>) and a media type
+(string <var>inputMediaType</var>) as inputs, and produces an object
+(<var>securingMechanismVerified</var>) that contains the following:
+        </p>
+
+        <ul>
+          <li>
+a verification status (boolean <var>status</var>)
+          </li>
+          <li>
+a verified document (object <var>document</var>)
+          </li>
+          <li>
+a media type (string <var>mediaType</var>)
+          </li>
+          <li>
+zero or more warnings (array of <a>ProblemDetails</a> <var>warnings</var>)
+          </li>
+          <li>
+zero or more errors (array of <a>ProblemDetails</a> <var>errors</var>)
+          </li>
+        </ul>
+
+        <p>
+The securing mechanism verification algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set <var>securingMechanismVerified</var> to an empty object and initialize the
+following fields: <var>status</var> to `false`, <var>warnings</var> to an
+empty array, and <var>errors</var> to an empty array.
+          </li>
+          <li>
+Determine the specific mechanism for extracting the <a>conforming document</a>
+(<var>extractDocument</var>):
+            <ol class="algorithm">
+              <li>
+If the <var>inputMediaType</var> is `application/vc+ld+json` or
+`application/vp+ld+json`, the <var>extractDocument</var> function takes
+<var>inputBytes</var> as an input parameter, parses the value as JSON,
+and returns the result as an object. If JSON parsing fails, a
+<a href="#PARSING_ERROR">PARSING_ERROR</a> is appended to
+<var>securingMechanismVerified</var>.<var>errors</var>.
+              </li>
+              <li>
+If the <var>inputMediaType</var> is associated with the JOSE or COSE
+specifications, the [[?VC-JOSE-COSE]] specification defines a process for
+extracting the <a>conforming document</a>.
+              </li>
+              <li>
+Other extracting functions are possible, but need not be listed in
+the <a data-cite="?VC-SPECS#media-type-extensions">Media Type Extensions</a>
+section of the Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+Implementers might find these other specifications through other standards
+bodies or communities that utilize this specification.
+              </li>
+            </ol>
+          </li>
+          <li>
+Determine the specific cryptographic verification algorithm
+(<var>verifyProof</var>) required to verify the securing mechanism:
+            <ol class="algorithm">
+              <li>
+If the <var>inputMediaType</var> is `application/vc+ld+json` or
+`application/vp+ld+json`, the Verifiable Credential Data Integrity
+Specification [[?VC-DATA-INTEGRITY]] defines the process for identifying
+a specific cryptography suite type used for cryptographic verification by
+utilizing the `proof`.`type` property. The list of all registered
```

Comment:
Fixed the link in 4aab6c96882676cb0a7a5aa75e1e92e9dc7260bb.

While there is a disparity between, in this case, the `byteSequence` (a string) and the `securedDocument` (an object)... it probably doesn't need to be spelled out that you `JSON.parse(string)` to get `securedDocument`.

This probably boils down to whether or not we need a strict interface between the VC spec and the securing specs, or if implementers can just figure it out since they're implementing that interface. IOW, we haven't had implementation complaints here... the implementers seem to be doing just fine figuring out stuff like that themselves.

The other wrinkle is that the Data Integrity specs just went into CR last week, so if we're going to change the function signatures now, it's going to kick us into another CR. We'll probably end up having to do another CR for different reasons, so changing the interfaces at this point is still possible if we can get the WG to agree to changing the interfaces.

All this said, I'll see what I can get done in this PR... and then in the group... we're probably going to have to create an issue and process it separately for the "interface" discussion since it impacts at least six specifications.